### PR TITLE
Trace NodeState: demo

### PR DESCRIPTION
**Pull request by denisshevchenko**
This is the first step toward `NodeState` tracing. The `NodeState` represents the current state of the node. So any external application (for example, `cardano-wallet` or `cardano-tracer`) will be able to ask this state (as a `DataPoint`) and use it.

Currently, `NodeState` is almost useless, it only demonstrates the changing the state of a node in real-time.

This PR introduces `demo-acceptor` as well: a tiny application allowing to ask `DataPoint` from the node.

### CODE REVIEW

**Review by deepfire — APPROVED** [review]
bors r+

File: cardano-node/src/Cardano/Node/StateRep.hs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE DeriveGeneric #-}
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/StateRep.hs:1]: Can we move this into the `Tracing` folder?

**Review by Jimbo4350 — COMMENTED** [review]
Can we also squash the first two commits?

File: cardano-node/src/Cardano/Node/StateRep.hs (hunk 1 above)
Comment (denisshevchenko) [reply cardano-node/src/Cardano/Node/StateRep.hs:1]: @Jimbo4350 Done.

**Comment by denisshevchenko** [comment]
@Jimbo4350 Done, now there are only 2 commits.

**Review by nc6 — APPROVED** [review]

**Comment by nc6** [comment]
bors merge

**Pull request merged**